docs/Guides/Migration-Guide-V4.md

```diff
@@ -65,6 +65,14 @@ Starting from v4, all the `GET` routes will create a sibling `HEAD` route.
 You can revert this behaviour by setting the server's option `exposeHeadRoutes`
 to `false`.
 
+### Synchronous route definitions
+
+The route registration has been made syncrounes from v4. This change was done to provide
+better error reporting for route definition. As a result if you specify an `onRoute` hook
+in a plugin you should either:
+* wrap your routes in a plugin (recommended)
+* use `await register(...)`
```

Comment:
I would add the code example to let the user understand better the implications:
any `onRoute` hooks will not be executed when a plugin adds them if it is not awaited or  the routes are wrapped in a plugin

Reply:
I added the code example used in the linked issue before and after refactoring. Could you have a look please if that was what you had in mind?